tests/components/plugwise/test_config_flow.py

```diff
@@ -84,8 +143,93 @@ async def test_form_cannot_connect(hass, mock_smile):
 
     result2 = await hass.config_entries.flow.async_configure(
         result["flow_id"],
-        {"host": "1.1.1.1", "password": "test-password"},
+        {"host": TEST_HOST, "password": TEST_PASSWORD},
     )
 
-    assert result2["type"] == "form"
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_FORM
     assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_other_problem(hass, mock_smile):
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    mock_smile.connect.side_effect = TimeoutError
+    mock_smile.gateway_id = "0a636a4fc1704ab4a24e4f7e37fb187a"
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {"host": TEST_HOST, "password": TEST_PASSWORD},
+    )
+
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": "unknown"}
+
+
+async def test_show_zeroconf_form(hass, mock_smile) -> None:
+    """Test that the zeroconf confirmation form is served."""
+    flow = config_flow.PlugwiseConfigFlow()
+    flow.hass = hass
+    flow.context = {"source": SOURCE_ZEROCONF}
+    result = await flow.async_step_zeroconf(TEST_DISCOVERY)
+
+    await hass.async_block_till_done()
+    assert flow.context["title_placeholders"][CONF_HOST] == TEST_HOST
+    assert flow.context["title_placeholders"]["name"] == "P1 DSMR v1.2.3"
+
+    assert result["step_id"] == "user"
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+
+
+async def test_options_flow_power(hass, mock_smile) -> None:
+    """Test config flow options DSMR environments."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        title=CONF_NAME,
+        data={CONF_HOST: TEST_HOST, CONF_PASSWORD: TEST_PASSWORD},
+        options={CONF_SCAN_INTERVAL: DEFAULT_SCAN_INTERVAL},
+    )
+
+    hass.data[DOMAIN] = {entry.entry_id: {"api": MagicMock(smile_type="power")}}
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.options.async_init(entry.entry_id)
```

Comment:
We should set up the config entry before starting the options flow.

The options flow requires the config flow module to be loaded. This will happen when setting up the config entry. If CI splits testing the config flow might not be loaded by the other tests when the options flow tests run.

Reply:
Tnx, will look into this and the below comment and get an PR started later today

Reply:
https://github.com/home-assistant/core/pull/39736 it is